apps/web/src/services/dashboards/widgets/_base/base-count-of-findings/BaseCountOfFindingsWidget.vue

```diff
@@ -158,8 +158,11 @@ const refineChartData = (data: Data[]): ChartData[] => {
     data.forEach((d) => {
         const fail_finding_count = d.value?.find((v) => v.key === 'fail_finding_count')?.value ?? 0;
         const pass_finding_count = d.value?.find((v) => v.key === 'pass_finding_count')?.value ?? 0;
+        const rawValue = d[state.groupByKey];
+        const referenceMap = Object.values(props.allReferenceTypeInfo ?? {}).find((info) => info.key === state.groupBy)?.referenceMap;
```

Comment:
suggestion: It would be much better if this line was executed above the `forEach` statement our outside of this function as `computed`.

Reply:
So useful comment!! Thanks!!!!!!!